### PR TITLE
Internal: Add atomic image [EDS-321]

### PR DESCRIPTION
**Pull request by IshayMaya**

### CODE REVIEW

File: modules/atomic-widgets/controls/types/select-control.php [hunk 1]
```diff
@@ -20,9 +20,13 @@ public function set_options( array $options ): self {
...
+	public function get_options() {
+		return $this->options;
+	}
+
 	public function get_props(): array {
 		return [
-			'options' => $this->options,
+			'options' => $this->get_options(),
 		];
 	}
```
Comment (blackcolt) [inline modules/atomic-widgets/controls/types/select-control.php:31]: Why not directly call $this->options;

File: modules/atomic-widgets/widgets/atomic-image.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+		// /** This filter is documented in wp-admin/includes/media.php */
+		return apply_filters( 'image_size_names_choose', $image_sizes );
+	}
+
+	protected function define_atomic_controls(): array {
+		$options = $this->get_image_sizes();
+
+		$resolution_control = Select_Control::bind_to( 'image_size' )
+			->set_label( __( 'Resolutions', 'elementor' ) )
```
Comment (rodolphebertozzo) [inline modules/atomic-widgets/widgets/atomic-image.php:90]: esc_html__

**Review by rodolphebertozzo — CHANGES_REQUESTED** [review]

File: modules/atomic-widgets/widgets/atomic-image.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+			->set_options( $options );
+
+		$content_section = Section::make()
+			->set_label( __( 'Content', 'elementor' ) )
```
Comment (blackcolt) [inline modules/atomic-widgets/widgets/atomic-image.php:94]: like @rodolphebertozzo said :)

File: modules/atomic-widgets/controls/types/select-control.php (hunk 1 above)
Comment (IshayMaya) [reply modules/atomic-widgets/controls/types/select-control.php:31]: @blackcolt needed the getter for tests so i changed it here as well

File: modules/atomic-widgets/controls/types/select-control.php [hunk 4]
```diff
@@ -20,9 +20,13 @@ public function set_options( array $options ): self {
...
 
+	public function get_options() {
+		return $this->options;
+	}
+
 	public function get_props(): array {
 		return [
-			'options' => $this->options,
+			'options' => $this->get_options(),
```
Comment (ManorHazaz) [inline modules/atomic-widgets/controls/types/select-control.php:29]: And remove the get_options()

```suggestion
			'options' => $this->options,
```

File: modules/atomic-widgets/widgets/atomic-image.php [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+		// TODO: Replace with actual URL prop
+		$image_url = $settings['url'];
+
+		// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+		echo "<img src='" . esc_url( $image_url ) . "' />";
+	}
+
+	private function get_image_sizes() {
+		$wp_image_sizes = self::get_all_image_sizes();
```
Comment (ManorHazaz) [inline modules/atomic-widgets/widgets/atomic-image.php:38]: Maybe?

```suggestion
	private function get_image_sizes_options() {
		$wp_image_sizes = self::get_wp_image_sizes();
```

**Review by ManorHazaz — CHANGES_REQUESTED** [review]

File: modules/atomic-widgets/widgets/atomic-image.php (hunk 5 above)
Comment (IshayMaya) [reply modules/atomic-widgets/widgets/atomic-image.php:38]: yep

**Review by ManorHazaz — APPROVED** [review]

**Pull request merged**